src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,159 @@ const ActorInit Dm_Hina_InitVars = {
     (ActorFunc)DmHina_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F470.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F56C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F63C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F75C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F9AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Draw.s")
+void DmHina_Init(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->isVisible = true;
+    this->actionFunc = func_80A1F470;
+    this->blueWarpPosY = this->actor.world.pos.y;
+    this->unk_148 = 0.0f;
+    this->unk_15C = 1.0f;
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.y = this->actor.world.pos.y;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
+}
+
+void DmHina_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80A1F470(DmHina* this, PlayState* play) {
+    Player* player;
+
+    player = GET_PLAYER(play);
+    Math_SmoothStepToF(&this->unk_148, 0.6f, 0.5f, 0.05f, 0.001f);
+    this->unk_154 = Math_SinS((play->gameplayFrames * 0x708)) * 8.0f;
```

Comment:
unk154 seems to be the bobbing timer, it controls how the model oscillates up and down in blue warp. 

(it might also control the rotation but I'm not as confident about my understanding of the matrix calculation there, but that too would make sense)

Reply:
I don't particularly understand what the matrix calculations are doing, so I will take your word on this